src/_nebari/upgrade.py

```diff
@@ -506,6 +618,12 @@ def _version_specific_upgrade(
 
 
 class Upgrade_2023_7_1(UpgradeStep):
+    """
+    Upgrade step for Nebari version 2023.7.1
```

Comment:
I don't think these docstrings do much. I would suggest leaving them out unless there are specific notes for that particular upgrade step.

Reply:
Have removed those docstrings having no specific notes